Description: Upstream changes introduced in version 7.3.50~cvs20110627-1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 gdb (7.3.50~cvs20110627-1) experimental; urgency=low
 .
   * Upstream pre-release based on CVS date 20110627.
   * Drop Daniel Jacobowitz as Maintainer.
       - Thanks for all this years.
   * Add myself as Maintainer.
   * Bump standards version.
   * Add local-option for unapply-patches.
   * Add prelink build depend, needed for tests run.
 .
 The person named in the Author field signed this changelog entry.
Author: Hector Oron <zumbi@debian.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

Index: gdb-7.5/gdb/main.c
===================================================================
--- gdb-7.5.orig/gdb/main.c	2012-07-20 17:59:05.000000000 +0000
+++ gdb-7.5/gdb/main.c	2012-11-12 20:27:57.000000000 +0000
@@ -846,7 +846,7 @@
      debugging or what directory you are in.  */
 
   if (home_gdbinit && !inhibit_gdbinit)
-    catch_command_errors (source_script, home_gdbinit, 0, RETURN_MASK_ALL);
+    catch_command_errors (source_script, home_gdbinit, -1, RETURN_MASK_ALL);
 
   /* Process '-ix' and '-iex' options early.  */
   for (i = 0; VEC_iterate (cmdarg_s, cmdarg_vec, i, cmdarg_p); i++)
@@ -950,7 +950,7 @@
 	{
 	  auto_load_local_gdbinit_loaded = 1;
 
-	  catch_command_errors (source_script, local_gdbinit, 0,
+	  catch_command_errors (source_script, local_gdbinit, -1,
 				RETURN_MASK_ALL);
 	}
     }
Index: gdb-7.5/gdb/cli/cli-cmds.c
===================================================================
--- gdb-7.5.orig/gdb/cli/cli-cmds.c	2012-11-12 20:23:53.000000000 +0000
+++ gdb-7.5/gdb/cli/cli-cmds.c	2012-11-12 20:27:03.000000000 +0000
@@ -1687,6 +1687,10 @@
 }
 
 
+/* Load a GDB command file whose name is given in ARGS.  FROM_TTY may
+   be -1, in which case we are loading a gdbinit file; in that case,
+   be paranoid about unsafe files.  */
+
 void
 init_cli_cmds (void)
 {
